### ISSUE TITLE
user.list_users does not work on smartos

### ISSUE THREAD

**Issue by Ch3LL** [saltstack/salt#34038]
### Description of Issue/Question
The excecution module `user.list_users` is not available in smartos and is not in solaris_user.py. 

### Setup
1. Minion on smartos

### Steps to Reproduce Issue
1.

```
[root@d860d7c7-1cb1-4b6a-9c83-5642a46b46c0 /opt/salt]# /opt/salt/bin/salt '*' user.list_users
d860d7c7-1cb1-4b6a-9c83-5642a46b46c0:
    'user.list_users' is not available.
ERROR: Minions returned with non-zero exit code
```

As shown above user.list_users is not able to run on smartos

ping @sjorge 

### Versions Report
Tested on 2016.3.0

**Comment by sjorge** [comment]
At first glance it looks like the one from useradd can just be copied to solaris_user.
Enabled my build host again, it should be online later tonight. I won't be home though so I will try and take a look later this week.

**Comment by rallytime** [comment]
@Ch3LL Should be fixed by #34042
